routes/v1.js

```diff
@@ -231,6 +301,60 @@ var patchAndValidateTaskDef = function(taskId, taskDef) {
   return null;
 };
 
+/** Ensure the taskGroup exists and that membership is declared */
+let ensureTaskGroup = async (ctx, taskId, taskDef, res) => {
+  let taskGroupId = taskDef.taskGroupId;
+  let taskGroup = await ctx.TaskGroup.load({taskGroupId}, true);
+  let taskGroupExpiration = new Date(
+    new Date(taskDef.expires).getTime() +
+    ctx.taskGroupExpiresExtension * 1000
+  );
+  if (!taskGroup) {
+    taskGroup = await ctx.TaskGroup.create({
+      taskGroupId,
+      schedulerId:  taskDef.schedulerId,
+      expires:      taskGroupExpiration,
+    }).catch(err => {
+      // We only handle cases where the entity already exists
+      if (!err || err.code !== 'EntityAlreadyExists') {
+        throw err;
```

Comment:
I think it treats `undefined` or `null` as any other value... You can do `throw "my string"` in javascript...
You shouldn't and I doubt we have code that do, but I wanted to avoid an error saying that `null doesn't have a property code`